contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -274,21 +276,113 @@ contract WithdrawalQueue is AccessControlEnumerable, WithdrawalQueueBase {
+    {
+        requestIds = new uint256[](_withdrawalRequestInputs.length);
+        for (uint256 i = 0; i < _withdrawalRequestInputs.length; ++i) {
+            uint256 amountOfWstETH = _withdrawalRequestInputs[i].amount;
+            address recipient = _checkWithdrawalRequestInput(
+                IWstETH(WSTETH).getStETHByWstETH(amountOfWstETH),
+                _withdrawalRequestInputs[i].recipient
+            );
+            requestIds[i] = _requestWithdrawalWstETH(amountOfWstETH, recipient);
+        }
+    }
+
+    struct Permit {
+        uint256 value;
+        uint256 deadline;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    /// @notice Request the sequence of stETH withdrawals according to passed `withdrawalRequestInputs` data using EIP-2612 Permit
+    /// @param _withdrawalRequestInputs an array of `WithdrawalRequestInput` data. The standalone withdrawal request will
+    ///  be created for each item in the passed list. If `WithdrawalRequestInput.recipient` is set to `address(0)`,
+    ///  `msg.sender` will be used as recipient.
+    /// @return requestIds an array of the created withdrawal requests
+    function requestWithdrawalBatchWithPermit(WithdrawalRequestInput[] memory _withdrawalRequestInputs, Permit memory _permit)
+        external
+        whenResumed
+        returns (uint256[] memory requestIds)
+    {
+        STETH.permit(msg.sender, address(this), _permit.value, _permit.deadline, _permit.v, _permit.r, _permit.s);
+        return requestWithdrawalBatchWstETH(_withdrawalRequestInputs);
+    }
+
+    /// @notice Request the sequence of wstETH withdrawals according to passed `withdrawalRequestInputs` data using EIP-2612 Permit
+    /// @param _withdrawalRequestInputs an array of `WithdrawalRequestInput` data. The standalone withdrawal request will
+    ///  be created for each item in the passed list. If `WithdrawalRequestInput.recipient` is set to `address(0)`,
+    ///  `msg.sender` will be used as recipient.
+    /// @return requestIds an array of the created withdrawal requests
+    function requestWithdrawalBatchWstETHWithPermit(WithdrawalRequestInput[] memory _withdrawalRequestInputs, Permit memory _permit)
+        external
+        whenResumed
+        returns (uint256[] memory requestIds)
+    {
+        WSTETH.permit(msg.sender, address(this), _permit.value, _permit.deadline, _permit.v, _permit.r, _permit.s);
+        return requestWithdrawalBatchWstETH(_withdrawalRequestInputs);
     }
 
     /// @notice Claim withdrawals batch once finalized (claimable)
-    /// NB: Always reverts
-    function claimWithdrawalBatch(uint256[] calldata /*_requests*/ ) external pure {
-        revert Unimplemented();
+    /// @param _requestIds ids of the finalized requests to claim
+    /// @param _hints rate indices found offchain that should be used for claiming
+    function claimWithdrawals(uint256[] calldata _requestIds, uint256[] calldata _hints) external {
```

Comment:
Imho, we should go with batches or with plurals for request and claim methods, but not mixed.

Reply:
Fixed. Removed non-batch versions of the methods for withdrawals requesting and claiming.